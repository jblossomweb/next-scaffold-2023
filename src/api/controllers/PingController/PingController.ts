import type { NextApiRequest, NextApiResponse } from 'next'

export interface Data {
  message: string
}

const PingController = (
  //
  req: NextApiRequest,
  res: NextApiResponse<Data>
) => {
  res.status(200).json({
    message: 'Pong',
  })
}

export default PingController
