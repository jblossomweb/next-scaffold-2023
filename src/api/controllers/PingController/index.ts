export { default, type Data } from './PingController'
