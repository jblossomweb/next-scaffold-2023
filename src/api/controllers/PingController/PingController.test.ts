/* eslint-disable import/no-extraneous-dependencies */
import { testApiHandler } from 'next-test-api-route-handler'
import PingController, { type Data } from './PingController'

describe('api/controllers/PingController', () => {
  it('should respond with expected data', async () => {
    await testApiHandler<Data>({
      handler: PingController,
      test: async ({ fetch }) => {
        const res = await fetch({ method: 'GET' })
        const data = await res.json()
        expect(res.status).toBe(200)
        expect(data).toMatchSnapshot()
      },
    })
  })
})
