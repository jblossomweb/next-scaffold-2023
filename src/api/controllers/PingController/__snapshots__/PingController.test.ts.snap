// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/controllers/PingController should respond with expected data 1`] = `
{
  "message": "Pong",
}
`;
